_posts/2019-07-08-persistent-storage-improvements.md

```diff
@@ -0,0 +1,261 @@
+For example:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta1
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  kafka:
+    # ...
+    storage:
+      type: jbod
+      volumes:
+      - id: 0
+        type: persistent-claim
+        size: 1000Gi
+        deleteClaim: false
+      - id: 1
+        type: persistent-claim
+        size: 1000Gi
+        deleteClaim: false
+# ...
+```
+
+If that is what you have, you can easily add more volumes just by editing the YAML.
+For example:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta1
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  kafka:
+    # ...
+    storage:
+      type: jbod
+      volumes:
+      - id: 0
+        type: persistent-claim
+        size: 1000Gi
+        deleteClaim: false
+      - id: 1
+        type: persistent-claim
+        size: 1000Gi
+        deleteClaim: false
+      - id: 10
+        type: persistent-claim
+        size: 1000Gi
+        deleteClaim: false
+      - id: 11
+        type: persistent-claim
+        size: 1000Gi
+        deleteClaim: false
+  # ...
+```
+
+Similarly you can also remove volumes:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta1
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  kafka:
+    # ...
+    storage:
+      type: jbod
+      volumes:
+      - id: 1
+        type: persistent-claim
+        size: 1000Gi
+        deleteClaim: false
+# ...
+```
+
+When removing volumes, Strimzi will not do anything to move the data from the volume you are removing.
+You have to do that manually before removing the volume(s) from the JBOD storage.
+Strimzi will simply remove the volume from the pods.
+
+**There are some things you should keep in mind and be careful about:**
+* The `id` numbers have to be unique.
+* The `id` numbers do not have to be in sequence.
+* **Changing the `id` number will be the same as removing the volume with the original `id` and adding a new volume with the new `id`. 
+Strimzi will stop using the old PVC with the old volume and create a new PVC for the new volume.**
+* By default the PVCs are not deleted. 
+So if you reuse an `id` which you already used in the past you should check first whether the old PVC still exists or not. 
+If it exists it will be reused instead of creating a new PVC with a new volume.
+
+## Using different storage class for each broker
+
+One of the other storage features we added in 0.12 release is to use different storage class for each broker.
+You can specify a different storage class for one or more Kafka brokers, instead of using the same storage class for all of them. 
+This is useful if, for example, storage classes are restricted to different availability zones or data centers. You can use the `overrides` field for this purpose.
+For example:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta1
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  kafka:
+    # ...
+    storage:
+      deleteClaim: true
+      size: 100Gi
+      type: persistent-claim
+      class: my-storage-class
```

Comment:
out of curiosity ... if I have a 3 brokers cluster and I want 3 different storage class (one for each broker) as described in this example, what happens if I don't set this "main" `class` field? Does it work? In this scenario, I don't need that because anyway I am going to override classes for each broker in the cluster. Of course, it means that if I increase the replicas and the `class` field isn't here something bad could happen. Just curiosity.

Reply:
The `class` field is basically the default. If you don't set it, the default storage class will be used as default (i.e. whatever is set as default in Kubernetes). That said, I would not go so deep into this in this post.